packages/graphql-connection-transformer/src/ModelConnectionTransformer.ts

```diff
@@ -46,6 +52,74 @@ function validateKeyField(field: FieldDefinitionNode): void {
     throw new InvalidDirectiveError(`If you define a field and specify it as a 'keyField', it must be of type 'ID' or 'String'.`)
 }
 
+/**
+ * Ensure that the field passed in is compatible to be a key field
+ * (Not a list and of type ID or String)
+ * @param field: the field to be checked.
+ */
+function validateKeyFieldNewConnection(field: FieldDefinitionNode): void {
+    if (!field) {
+        return
+    }
+    const isNonNull = isNonNullType(field.type);
+    const isAList = isListType(field.type)
+
+    // The only valid key fields are single non-null fields.
+    if (!isAList && isNonNull) {
+        return;
+    }
+    throw new InvalidDirectiveError(`All fields provided to an @connection must be non-null scalar fields.`)
+}
+
+/**
+ * Checks that the fields being used to query match the expected key types for the index being used.
+ * @param parentFields: All fields of the parent object.
+ * @param realtedTypeFields: All fields of the related object.
+ * @param inputFieldNames: The fields passed in to the @connection directive.
+ * @param keySchema: The key schema for the index being used.
+ */
+function checkFieldsAgainstIndex(parentFields: ReadonlyArray<FieldDefinitionNode>,
+                                 relatedTypeFields: ReadonlyArray<FieldDefinitionNode>,
+                                 inputFieldNames: string[],
+                                 keySchema: KeySchema[]): void {
+    const hashAttributeName = keySchema[0].AttributeName;
+    const tablePKType = relatedTypeFields.find(f => f.name.value === hashAttributeName).type;
```

Comment:
Can we've a situation when ```hashAttributeName``` cannot be found within the fields? because there is no check for the result of ```find```, just the result used straight ahead.